sequana/ribodesigner.py

```diff
@@ -0,0 +1,133 @@
+
+
+def extract_regions_from_fasta(fasta_in, regions_df, fasta_out):
+    """From a genomic FASTA file fasta_in, extract regions stored in regions_df
+    dataframe obtained with get_rna_pos_from_gff. Export the corresponding
+    regions to a FASTA file fasta_out.
+
+    :param fasta_in: path to genomic FASTA file to extract rRNA regions from.
+    :param regions_df: pandas DataFrame generated by get_rna_pos_from_gff
+    :param fasta_out: path to FASTA file where rRNA regions will be exported to.
+    """
+
+    count = 0
+
+    with pysam.Fastafile(fasta_in) as fas:
+        with open(fasta_out, "w") as fas_out:
+            for rna in regions_df.to_dict("records"):
+                region = f"{rna['seqid']}:{rna['start']}-{rna['end']}"
+                seq_record = f">{rna['seqid']}:{rna['start']}-{rna['end']}\n{fas.fetch(region=region)}\n"
+                fas_out.write(seq_record)
+                count += 1
+
+    logger.info(f"{count} regions were extracted from '{fasta_in}' to '{fasta_out}'")
+
+
+def get_probes(fasta_in, fasta_out, probe_len=50, inter_probe_space=15):
+    """From a FASTA file of nucleotide sequence to be targetted, generate a FASTA
+    file with probe designed of size 'probe_len' and with a space between probes
+    of 'inter_probe_space'
+
+    :param fasta_in: path to FASTA file with extracted rRNA regions.
+    :param fasta_out: path to FASTA file where probes will be exported to.
+    :param prob_len: the size for probes in nucleotides.
+    :param inter_probe_space: the space between probes in nucleotides.
+    """
+
+    count = 0
+    with open(fasta_out, "w") as fas_out:
+
+        with pysam.FastxFile(fasta_in) as fas:
+            for seq in fas:
+
+                for start in range(0, len(seq.sequence) - probe_len, probe_len + inter_probe_space):
+                    count += 2
+
+                    end = start + probe_len
+
+                    probe_fw = f">{seq.name}:probe_{start}_{end}_forward\n{seq.sequence[start:end]}\n"
+                    rev_seq = reverse_complement(seq.sequence)
+                    probe_rv = f">{seq.name}:probe_{start}_{end}_reverse\n{rev_seq[start:end]}\n"
+
+                    fas_out.write(probe_fw + probe_rv)
+
+    logger.info(f"{count} probes designed.")
+
+
+def cluster_probes(fasta_in, fasta_out, seq_id_thres=0.80, threads=4):
+    """Use cd-hit-est to cluster highly similar probes
+
+    :param fasta_in: path to FASTA file with probes.
+    :param fasta_out: path to FASTA file with probes after filtering.
+    """
+    log_file = Path(fasta_out).with_suffix(".log")
+
+    cmd = f"cd-hit-est -i {fasta_in} -o {fasta_out} -c {seq_id_thres} -n {threads}"
+    logger.info(f"Clustering probes with command: {cmd} (log in '{log_file}').")
+
+    with open(log_file, "w") as f:
+        subprocess.run(cmd, shell=True, check=True, stdout=f)
+
+
+def fasta_to_csv(fasta_in, csv_out):
+    """Convert a FASTA file into a CSV file with the oligo name in column 1 and the
```

Comment:
Convert a FASTA file into a CSV file 

The the oligo name in column 1 and the oligo sequence in column 2